actions/qase/qase.go

```diff
@@ -0,0 +1,195 @@
+	} else if resultLength == 1 {
+		return &qaseSuites.Result.Entities[0], nil
+	}
+
+	return nil, fmt.Errorf("test suite \"%s\" not found in project %s", suite, project)
+}
+
+// CreateTestSuite creates a new Test Suite within a specified Qase Project
+func (q *Service) CreateTestSuite(project string, suite upstream.SuiteCreate) (int64, error) {
+	logrus.Debugf("Creating test suite \"%s\" in project %s\n", suite.Title, project)
+	resp, _, err := q.Client.SuitesApi.CreateSuite(context.TODO(), suite, project)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create test suite: \"%s\". Error: %v", suite.Title, err)
+	}
+	return resp.Result.Id, nil
+}
+
+// createTestCase creates a new test in qase
+func (q *Service) createTestCase(project string, testCase upstream.TestCaseCreate) error {
+	_, _, err := q.Client.CasesApi.CreateCase(context.TODO(), testCase, project)
+	if err != nil {
+		return fmt.Errorf("failed to create test case: \"%s\". Error: %v", testCase.Title, err)
+	}
+	return nil
+}
+
+// updateTestCase updates an existing test in qase
+func (q *Service) updateTestCase(project string, testCase upstream.TestCaseUpdate, id int32) error {
+	_, _, err := q.Client.CasesApi.UpdateCase(context.TODO(), testCase, project, id)
+	if err != nil {
+		return fmt.Errorf("failed to update test case: \"%s\". Error: %v", testCase.Title, err)
+	}
+	return nil
+}
+
+// createSuitePath creates a series of nested test suites from a / deliniated string
+func createSuitePath(client *Service, suiteName, project string) (int64, error) {
+	suites := strings.Split(suiteName, "/")
+	testSuiteId := int64(0)
+	var parentID *int64
+	for _, suite := range suites {
+		testSuite, err := client.GetTestSuite(project, suite, parentID)
+		if testSuite != nil {
+			testSuiteId = testSuite.Id
+		}
+
+		if err != nil && testSuite != nil {
+			logrus.Error("Could not determine test suite:", err)
+			return 0, err
+		} else if err != nil {
+			logrus.Debug("Error obtaining test suite:", err)
+			suiteBody := upstream.SuiteCreate{Title: suite}
+			if testSuiteId != 0 {
+				suiteBody.ParentId = testSuiteId
+			}
+			testSuiteId, _ = client.CreateTestSuite(project, suiteBody)
+		}
+
+		parentID = &testSuiteId
+	}
+
+	return testSuiteId, nil
+}
+
+// UploadTests either creates new Test Cases and their associated Suite or updates them if they already exist
+func (q *Service) UploadTests(project string, testCases []upstream.TestCaseCreate) error {
+	for _, tc := range testCases {
+		existingCase, err := q.getTestCase(project, tc)
+		if err == nil {
+			logrus.Info("Updating test case:\n\tProject: ", project, "\n\tTitle: ", tc.Title, "\n\tSuiteId: ", tc.SuiteId)
+			var qaseTest upstream.TestCaseUpdate
+			qaseTest.Title = tc.Title
+			qaseTest.SuiteId = tc.SuiteId
+			qaseTest.Description = tc.Description
+			qaseTest.Type_ = tc.Type_
+			qaseTest.Priority = tc.Priority
+			qaseTest.IsFlaky = tc.IsFlaky
+			qaseTest.Automation = int32(2)
```

Comment:
There was a nice update to make this not always say automated -- can we add this?

ref: https://github.com/rancher/tests/pull/132/files#diff-1b2c3b1e83b3ebea187affbb7a36cfc882e7866c791de1692ea199f017de4811